buildSrc/src/main/java/org/elasticsearch/gradle/util/GradleUtils.java

```diff
@@ -61,7 +61,8 @@ public static SourceSetContainer getJavaSourceSets(Project project) {
     }
 
     public static <T> T maybeCreate(NamedDomainObjectContainer<T> collection, String name) {
-        return Optional.ofNullable(collection.findByName(name)).orElse(collection.create(name));
+        return collection.maybeCreate(name);
```

Comment:
Seems we don't really need this method then.

Reply:
removed all of those